qiskit_experiments/library/characterization/t2ramsey.py

```diff
@@ -103,20 +103,26 @@ def circuits(self, backend: Optional[Backend] = None) -> List[QuantumCircuit]:
             The experiment circuits
 
         Raises:
-            AttributeError: if unit is 'dt', but 'dt' parameter is missing in the backend configuration
+            AttributeError: if unit is 'dt', but 'dt' parameter is missing in the backend configuration.
+            ValueError: if values in `delays` are not strictly increasing.
         """
         if self._unit == "dt":
             try:
                 dt_factor = getattr(backend._configuration, "dt")
             except AttributeError as no_dt:
                 raise AttributeError("Dt parameter is missing in backend configuration") from no_dt
+        if not all(d1 < d2 for d1, d2 in zip(self._delays, self._delays[1:])):
+            raise ValueError("Delays should be strictly increasing")
 
         circuits = []
+        min_delay = self._delays[0] if self._delays[0] != 0 else self._delays[1]
         for delay in self._delays:
             circ = qiskit.QuantumCircuit(1, 1)
             circ.h(0)
             circ.delay(delay, 0, self._unit)
-            circ.p(2 * np.pi * self._osc_freq, 0)
+            nosc = delay / min_delay
+            rotation_angle = 2 * np.pi * self._osc_freq * nosc
+            circ.p(rotation_angle, 0)
```

Comment:
A few things:
1. I would prefer we use an `rz` gate for this. (I realize they are equivalent up to a global phase, but the `rz` really makes it explicit what you are doing.
2. In NMR they call they call this a "time proportional phase increment" (TPPI). It may be worthwhile to use that term in the doc string for the class for users more familiar with that terminology.
3. The rotation angle you want is in terms of absolute time and not relative delay, i.e.:
```python
delay_seconds = convert_unit(delay, self._unit, "s")
circ.rz(2 * np.pi * self._osc_freq * delay_seconds)
```
(assuming the existence of a `convert_unit` method that is more generic than the `convert_unit` method in terra)

With this change you can drop the book keeping about `min_delay`.

Reply:
@blakejohnson , thanks for the comments. I updated (1) and (3) as you suggested. It certainly makes more sense now. I will update the documentation (2) after the release.
@chriseclectic - once the tests pass (hopefully), can you review once more?